lib/puppet/configurer.rb

```diff
@@ -472,6 +433,25 @@ def find_functional_server
   end
   private :find_functional_server
 
+  def last_used_environment
+    if Puppet::FileSystem.exist?(Puppet[:lastrunfile])
+      begin
+        last_run_environment = Puppet::Util::Yaml.safe_load_file(Puppet[:lastrunfile])['version']['environment']
```

Comment:
Maybe use dig here (ruby 2.3 and up supports it)?

Reply:
I wanted to make sure we also check last_run_report.yaml if the summary is invalid YAML or an empty file (not sure how much of a concern this is, since we would overwrite it at the end of the puppet run).

Reply:
I ended up using dig to ensure we don't raise on existing files that don't have the `environment` key (high chance), and also rescuing any exception (for when the file is invalid/empty - low chance).